### PR TITLE
Added common code module project and moved across clean architecture and mapper interfaces

### PR DESCRIPTION
**Pull request by spanersoraferty**
I've obviously had to move across a lot files to update the dependencies, whilst in there I've also done the namespace shortening that Cath likes (for consistency).

### CODE REVIEW

File: Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [Fact]
+    public void UseCase_HandleRequest_No_Args_ReturnsExpectedUseCaseReponse()
+    {
+        // arrange
+        var useCaseResponseMessage =
+            UseCaseTestDoubles.DefaultUseCaseResponseMessage;
+        var useCaseResponsefake =
+            UseCaseTestDoubles.DefaultUseCaseResponseFake(useCaseResponseMessage);
+        IUseCase<ExpandoObject> useCaseMock =
+            UseCaseTestDoubles.MockUseCaseFor(useCaseResponsefake);
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs:20]: I don't understand what these tests are for, mocking the UseCase

Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs:20]: So we can test the interface contract

Comment (CathLass) [reply Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs:20]: It's just an interface contract though - there's nothing to test?

Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs:20]: Well it's contractural behaviour so some advocate for testing, I'm not too bothered cos we have concrete implementations we test against these contracts in this solution, the component library didn't.

Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/Common/Tests/CleanArchitecture/Application/UseCase/UseCaseTests.cs:20]: Tis gone

**Review by CathLass — APPROVED** [review]

**Pull request merged**